packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -418,26 +423,53 @@ class _DragAvatar<T> extends Drag {
     _entry.markNeedsBuild();
     HitTestResult result = new HitTestResult();
     WidgetFlutterBinding.instance.hitTest(result, globalPosition + feedbackOffset);
-    _DragTargetState<T> target = _getDragTarget(result.path);
-    if (target == _activeTarget)
+
+    Iterable<_DragTargetState<T>> targets = _getDragTargets(result.path);
+
+    bool listsMatch = false;
+    if (targets.length >= _lastTargets.length && _lastTargets.isNotEmpty) {
```

Comment:
every time you do anything on `targets`, you're rerunning the entire hitTest loop, which is very expensive. Maybe call `.toList()` on the output of `_getDragTargets`?